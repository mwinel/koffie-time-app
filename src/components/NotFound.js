import React, { Component } from "react";

class NotFound extends Component {
  render() {
    return (
      <div className="container">
        <div className="row">
          <div className="col-10 mx-auto text-center text-title text-uppercase pt-5">
            <h1>404</h1>
            <h4>page not found</h4>
          </div>
          <div className="col-10 mx-auto text-center">
            <h5>
              The requested URL{" "}
              <span className="text-danger">
                {this.props.location.pathname}
              </span>{" "}
              was not found.
            </h5>
          </div>
        </div>
      </div>
    );
  }
}

export default NotFound;
