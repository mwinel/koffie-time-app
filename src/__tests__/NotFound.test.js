import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";

import NotFound from "../components/NotFound";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const location = { pathname: '/notfound' };

it("renders notfound component", () => {
  act(() => {
    render(<NotFound location={location} />, container);
  });
  expect(container.textContent).toBe("404page not foundThe requested URL /notfound was not found.");
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
