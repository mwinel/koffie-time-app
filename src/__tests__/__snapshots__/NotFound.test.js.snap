// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders notfound component 1`] = `
"<div class=\\"container\\">
  <div class=\\"row\\">
    <div class=\\"col-10 mx-auto text-center text-title text-uppercase pt-5\\">
      <h1>404</h1>
      <h4>page not found</h4>
    </div>
    <div class=\\"col-10 mx-auto text-center\\">
      <h5>The requested URL <span class=\\"text-danger\\">/notfound</span> was not found.</h5>
    </div>
  </div>
</div>"
`;
